api/restHandler/app/BuildPipelineRestHandler.go

```diff
@@ -1053,6 +1055,66 @@ func (handler PipelineConfigRestHandlerImpl) GetCIPipelineById(w http.ResponseWr
 	common.WriteJsonResp(w, err, ciPipeline, http.StatusOK)
 }
 
+func (handler PipelineConfigRestHandlerImpl) GetCIPipelineByPipelineId(w http.ResponseWriter, r *http.Request) {
+	token := r.Header.Get("token")
+	var ciPipelineId int
+	var err error
+	v := r.URL.Query()
+	pipelineId := v.Get("pipelineId")
+	if len(pipelineId) != 0 {
+		ciPipelineId, err = strconv.Atoi(pipelineId)
+		if err != nil {
+			handler.Logger.Errorw("request err, GetCIPipelineByPipelineId", "err", err, "pipelineIdParam", pipelineId)
+			response.WriteResponse(http.StatusBadRequest, "please send valid pipelineId", w, errors.New("pipelineId id invalid"))
+			return
+		}
+	}
+
+	handler.Logger.Infow("request payload, GetCIPipelineByPipelineId", "pipelineId", pipelineId)
+
+	ciPipeline, err := handler.pipelineBuilder.GetCiPipelineById(ciPipelineId)
+	if err != nil {
+		handler.Logger.Infow("service error, GetCIPipelineById", "err", err, "pipelineId", pipelineId)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	app, err := handler.pipelineBuilder.GetApp(ciPipeline.AppId)
+	if err != nil {
+		handler.Logger.Infow("service error, GetCIPipelineByPipelineId", "err", err, "appId", ciPipeline.AppId, "pipelineId", pipelineId)
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+	ciPipeline.AppName = app.AppName
+	ciPipeline.AppType = app.AppType
+
+	resourceName := handler.enforcerUtil.GetAppRBACName(app.AppName)
```

Comment:
not optimized, instead use method `GetAppRBACNameByAppId`

Reply:
sure have used